workbench/_api/routes/causal_mediation.py

```
@@ -0,0 +1,250 @@
+    results = backend()
+
+    # The model can be deregistered from the catalog (NDIF stopped serving it)
+    # between /start and /results; state[...] raises KeyError in that case.
+    # Surface a clear 503 instead of an opaque 500.
+    try:
+        model = state[req.model]
+    except KeyError:
+        raise HTTPException(
+            status_code=503,
+            detail=f"Model {req.model} is no longer available; please re-run.",
+        )
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)